session/ephemeral.go

```diff
@@ -0,0 +1,57 @@
+package session
```

Comment:
I don't think this belongs in package `session`. I actually think it belongs in its own package. Perhaps a `util` package since it us general purpose.